internal/packager/registry.go

```diff
@@ -128,11 +129,39 @@ func Push(appname, namespace, tag string) error {
 
 // Pull pulls an app from a registry
 func Pull(repotag string) error {
+	if err := pullImage(repotag); err != nil {
+		return err
+	}
+	return Load(repotag, ".")
+}
+
+func pullImage(repotag string) error {
 	cmd := exec.Command("docker", "pull", repotag)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	if err := cmd.Run(); err != nil {
 		return errors.Wrapf(err, "error pulling image %s", repotag)
 	}
-	return Load(repotag, ".")
+	return nil
+}
+
+type imageComponents struct {
+	Name       string
+	Repository string
+	Tag        string
+}
+
+func splitImageName(repotag string) (*imageComponents, error) {
+	named, err := reference.ParseNormalizedNamed(repotag)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse image name")
+	}
+	res := &imageComponents{
+		Repository: named.Name(),
+	}
+	res.Name = res.Repository[strings.LastIndex(res.Repository, "/")+1:]
```

Comment:
What if the last character of `res.Respository` is `"/"` ?

Reply:
That wouldn't be a valid repository identifier and would cause `ParseNormalizedName()` above to return an error. I'll add it as a test-case to ensure it behaves according to expectations.